### PR TITLE
Implement `inspect_item` and `inspect_url`

### PR DESCRIPTION
**Pull request by somespecialone**
I hope my code good enough. Also, i don't know why decoding of wear wont work(and don't understand what the `utils.StructIO` should do). Throws error:
```sh
Traceback (most recent call last):
  File "...\steam-ext-csgo\steam\utils.py", line 732, in maybe_coroutine
    value = func(*args, **kwargs)
  File "...\steam-ext-csgo\steam\ext\csgo\state.py", line 230, in handle_client_preview_data_block_response
    item.paint_wear = io.read_f32()
  File "<string>", line 1, in read_f32
  File "...\steam-ext-csgo\steam\utils.py", line 601, in read_struct
    return struct.unpack(format, buffer)
struct.error: unpack requires a buffer of 4 bytes
```
Tested on windows 10 x64, python 3.10; docker image `python3:10-slim-buster` and `python3:10` - same result. So i simply fix it by use `struct.pack/unpack`. I decided to return `protobufs.cstrike.PreviewDataBlock` from inspect (similar to [DoctorMcKay csgo inspectItem](https://github.com/DoctorMcKay/node-globaloffensive#inspectitemowner-assetid-d-callback)) and i really would be tnank if you point me out if i did something wrong

### CODE REVIEW

File: steam/ext/csgo/backpack.py [hunk 1]
```diff
@@ -110,12 +110,21 @@ async def contents(self) -> list[ProtoItem]:
...
-    ):
-        ...
+    @property
+    def inspect_url(self) -> str | None:
+        """
+        Get inspect url of item if it `inspectable`
+        :return: Inspect url or None
+        """
+        try:
+            for action in self.actions or []:
```
Comment (Gobot1234) [inline steam/ext/csgo/backpack.py]: this or is unnecessary
```suggestion
            for action in self.actions:
```

**Comment by Gobot1234** [comment]
I would like to keep the inspect method of BackpackItem and just make it call `self._state.client.inspect_item(self.inspect_url)`

**Comment by somespecialone** [comment]
> I would like to keep the inspect method of BackpackItem and just make it call `self._state.client.inspect_item(self.inspect_url)`

So, why we need this? Items in our backpack is already inspected. Other users items can be `BackpackItem` class instance? I don't get it...

**Comment by Gobot1234** [comment]
The constructor for BackpackItem is meant to be public so you can access things like def_index without having to figure out the schema yourself (this isn't implemented for this module yet, but it is for ext-tf2)

File: steam/ext/csgo/client.py [hunk 2]
```diff
@@ -58,6 +62,71 @@ async def _handle_ready(self) -> None:
...
+            a: int = 0,
+            d: int = 0,
+            m: int = 0,
+            url: str = "",
+            timeout: int=10
+    ) -> PreviewDataBlock:
+        """
+        The parameters can be taken from `inspect` links either from an inventory or market.
+        The market has the `m` parameter, while the inventory one has `s`.
+        :param s: steam id64 of owner
```
Comment (Gobot1234) [inline steam/ext/csgo/client.py]: I'd rather have the user not pass this, s can just be removed, they should pass owner.

File: steam/ext/csgo/client.py [hunk 3]
```diff
@@ -58,6 +62,71 @@ async def _handle_ready(self) -> None:
...
+        if s == 0 and m == 0:
+            raise TypeError(f"Missing required keyword-only argument: {'s' if not s else 'm'}")
+
+        params = {
+            'param_s': s,
+            'param_a': a,
+            'param_d': d,
+            'param_m': m,
+        }
+        await self.ws.send_gc_message(GCMsgProto(Language.Client2GCEconPreviewDataBlockRequest, **params))
```
Comment (Gobot1234) [inline steam/ext/csgo/client.py]:
```suggestion
        await self.ws.send_gc_message(
            GCMsgProto(
                    Language.Client2GCEconPreviewDataBlockRequest,
                    param_s=s,
                    param_a=a
                    param_d=d,
                    param_m=m,
                )
        )
```
This reads much better

File: steam/ext/csgo/state.py [hunk 4]
```diff
@@ -216,10 +216,9 @@ def handle_players_profile(self, msg: GCMsgProto[cstrike.PlayersProfile]):
     @register(Language.Client2GCEconPreviewDataBlockResponse)
     def handle_client_preview_data_block_response(self, msg: GCMsgProto[cstrike.Client2GcEconPreviewDataBlockResponse]):
         # decode the wear
-        with utils.StructIO() as io:
-            io.write_u32(msg.body.iteminfo.paintwear)
-            item = utils.get(self.backpack, id=msg.body.iteminfo.itemid)
-            item.paint_wear = io.read_f32()
+        item = msg.body.iteminfo
```
Comment (Gobot1234) [inline steam/ext/csgo/state.py:219]: This isn't used
```suggestion
```

File: steam/ext/csgo/client.py [hunk 5]
```diff
@@ -58,6 +62,71 @@ async def _handle_ready(self) -> None:
         self.http.user = ClientUser(self._connection, await self.http.get_user(self.user.id64))
         await super()._handle_ready()
 
+    @overload
+    async def inspect_item(self, *, owner: Type[SteamID], asset_id: int, d: int) -> PreviewDataBlock:
```
Comment (Gobot1234) [inline steam/ext/csgo/client.py]:
```suggestion
    async def inspect_item(self, *, owner: SteamID, asset_id: int, d: int) -> PreviewDataBlock:
```
This type hint is wrong

File: steam/ext/csgo/state.py (hunk 4 above)
Comment (somespecialone) [reply steam/ext/csgo/state.py:219]: I can't commit that - item variable is used in next few lines of code. Do you mean i need change item variable to msg.body.iteminfo?
```python
  packed_wear = struct.pack(">l", msg.body.iteminfo.paintwear)
  msg.body.iteminfo.paintwear = struct.unpack(">f", packed_wear)[0]
  self.dispatch("inspect_item_info", msg.body.iteminfo)
```

Comment (Gobot1234) [reply steam/ext/csgo/state.py:219]: I'm actually going to add an inspected item class or something so people can get more attributes like position and have a similar structure to the other classes I have in the works

File: steam/ext/csgo/client.py [hunk 6]
```diff
@@ -58,6 +62,64 @@ async def _handle_ready(self) -> None:
...
+
+    async def inspect_item(
+            self,
+            *,
+            owner: SteamID | None = None,
+            asset_id: int = 0,
+            d: int = 0,
+            market_id: int = 0,
+            url: str = "",
+            _timeout: int = 10
```
Comment (Gobot1234) [inline steam/ext/csgo/client.py]: Why is this argument here if it's completely hidden?

**Review by Gobot1234 — CHANGES_REQUESTED** [review]
This is pretty much there now

File: steam/ext/csgo/client.py (hunk 6 above)
Comment (somespecialone) [reply steam/ext/csgo/client.py]: I just left him if user need more control over waiting timeout. Not sure, if this is ever come handy

**Comment by somespecialone** [comment]
I think 60 seconds is too much, couse if we send wrong data to GC it just not respond, so we wait 60 seconds till `asyncio.TimeoutError` raised. Anyway, thank you for feedback!

**Comment by Gobot1234** [comment]
> I think 60 seconds is too much, couse if we send wrong data to GC it just not respond, so we wait 60 seconds till `asyncio.TimeoutError` raised. Anyway, thank you for feedback!

If someone wants this to be shorter they can just use asyncio.wait_for themselves.

**Comment by Gobot1234** [comment]
> I think 60 seconds is too much, couse if we send wrong data to GC it just not respond, so we wait 60 seconds till `asyncio.TimeoutError` raised. Anyway, thank you for feedback!

If someone wants this to be shorter they can just use asyncio.wait_for themselves.

**Comment by Gobot1234** [comment]
Thanks for this, I'll fix up the formatting changes on my end as restyler seems bugged.

**Pull request merged**